src/main/java/mk/edu/uklo/fikt/fiktexamweb/security/AuthenticationHandler.java

```diff
@@ -0,0 +1,87 @@
+package mk.edu.uklo.fikt.fiktexamweb.security;
+
+import mk.edu.uklo.fikt.fiktexamweb.util.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.web.DefaultRedirectStrategy;
+import org.springframework.security.web.RedirectStrategy;
+import org.springframework.security.web.WebAttributes;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.Collection;
+
+public class AuthenticationHandler implements AuthenticationSuccessHandler {
+
+    @Autowired
+    private UserService userService;
+
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Authentication authentication) throws IOException, ServletException {
+        handle(httpServletRequest, httpServletResponse, authentication);
+        clearAuthenticationAttributes(httpServletRequest);
+    }
+
+    private RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+
+    protected String determineTargetUrl(Authentication authentication) {
+        boolean isStudent = false;
+        boolean isTeacher = false;
+        boolean isAdmin = false;
+        Collection<? extends GrantedAuthority> authorities
+                = authentication.getAuthorities();
+        for (GrantedAuthority grantedAuthority : authorities) {
+            if (grantedAuthority.getAuthority().equals("Student")) {
+                isStudent = true;
+                break;
+            } else if (grantedAuthority.getAuthority().equals("Admin")) {
+                isAdmin = true;
+                break;
+            } else if (grantedAuthority.getAuthority().equals("Teacher")) {
+                isTeacher = true;
+                break;
+            }
+        }
+
+        int userId = userService.getIdByUsername(authentication.getName());
+
+        if (isStudent) {
+            return "/subject/studentsubjectscreen/";
```

Comment:
Extract paths in constants and use them everywhere where needed, like in controllers. That way if there is a need for a path to be changed it will be done on one place and will not be forgotten to be updated on all places